assets/javascripts/discourse/models/assignment.js

```diff
@@ -0,0 +1,38 @@
+import { tracked } from "@glimmer/tracking";
+import EmberObject from "@ember/object";
+
+export class Assignment extends EmberObject {
```

Comment:
I've introduced this model on the client, but I haven't migrated all code to using it because that would extend the refactoring surface too much. It would be great to slowly migrate all code to using it, I think that will help simplify the code of the plugin. Now it contains only fields, but as soon as more code start using it, it will become possible to move some domain logic in here.